src/shared_gui_components/EditView.cpp

```diff
@@ -400,4 +402,80 @@ void InputCheckBox::setIncomplete(bool incomplete) {
   }
 }
 
+PathInputView::PathInputView(const std::string& extension, bool isRead)
+  : InputView(), lineEdit(new QLineEdit()), nameLabel(new QLabel()), m_isRead(isRead) {
+  auto* vLayout = new QVBoxLayout();
+  vLayout->setContentsMargins(0, 0, 0, 0);
+  vLayout->setSpacing(5);
+  setLayout(vLayout);
+
+  nameLabel->setOpenExternalLinks(true);
+  nameLabel->setWordWrap(true);
+  vLayout->addWidget(nameLabel);
+
+  auto* hLayout = new QHBoxLayout();
+  hLayout->addWidget(lineEdit);
+
+  selectPathButton = new QPushButton("...");
+  selectPathButton->setFlat(true);
+  selectPathButton->setFixedSize(30, 20);
+  selectPathButton->setObjectName("StandardGrayButton");
+
+  connect(selectPathButton, &QPushButton::clicked, this, &PathInputView::onSelectPathButtonClicked);
+  hLayout->addWidget(selectPathButton);
+
+  vLayout->addLayout(hLayout);
+
+  // TODO: sanitize the input?
+  m_extension = toQString(extension);
+}
+
+void PathInputView::setName(const std::string& name, const boost::optional<std::string>& units, const boost::optional<std::string>& description) {
+  QString text;
+  text += QString::fromStdString(name);
+  if (units) {
+    text += QString::fromStdString(" (" + units.get() + ")");
+  }
+  if (description) {
+    text += QString::fromStdString("<div style=\"font-size:small;margin-top:2px;\">" + description.get() + "</div>");
+  }
+
+  nameLabel->setText(text);
+}
+
+void PathInputView::setIncomplete(bool incomplete) {
+  if (incomplete) {
+    nameLabel->setStyleSheet("QLabel { color: #DD0A05;}");
+  } else {
+    nameLabel->setStyleSheet("QLabel { color: black;}");
+  }
+}
+
+void PathInputView::setDisplayValue(const QVariant& value) {
+  lineEdit->setText(value.toString());
+}
+
+void PathInputView::onSelectPathButtonClicked() {
+  QString lastPath = lineEdit->text();
+
+  QString selectedPath;
+
+  if (m_extension.isEmpty()) {
```

Comment:
This logic seems reasonable, should we always append "Any files (*)" to the extension?  The docs for getSaveFileName say:

> Multiple filters are separated with ';;'

`.

Reply:
No extension == directory is the only kind of fishy logic here, that should really be an attribute of the argument itself (`isDir` like `isRead`), if you are updating the SDK maybe this is worth adding?

Reply:
Yes, for sure need some work: https://github.com/NREL/OpenStudio/issues/5273